lib/beaker/hypervisor/hcloud.rb

```diff
@@ -15,10 +20,13 @@ def initialize(hosts, options) # rubocop:disable Lint/MissingSuper
       @options = options
       @logger = options[:logger] || Beaker::Logger.new
       @hosts = hosts
+      @delete_vm_after = ENV.fetch('BEAKER_HCLOUD_DELETE_VM_AFTER', 1).to_i
 
+      raise 'BEAKER_HCLOUD_DELETE_VM_AFTER needs to be a positive integer' unless @delete_vm_after.positive?
       raise 'You need to pass a token as BEAKER_HCLOUD_TOKEN environment variable' unless ENV['BEAKER_HCLOUD_TOKEN']
 
       @client = ::Hcloud::Client.new(token: ENV.fetch('BEAKER_HCLOUD_TOKEN'))
+      delete_servers_if_required
```

Comment:
Doing this in the constructor is dangerous. It means you can't really write unit tests.

Why don't you add this to the cleanup? There is a risk that if things crash you never get to the cleanup, but you can also consider writing a specific clean up only task that runs X times a day in some common repository.

Reply:
I considered that. maybe we just run a github action as cron somewhere that does the cleanup. That also fixes the problems of cuncurrent cleanup jobs